File: app/src/main/java/ch/bailu/aat/gpx/GpxFileWrapper.java
package ch.bailu.aat.gpx;

import ch.bailu.simpleio.foc.Foc;


public class GpxFileWrapper extends GpxInformation {
    private final GpxList list;
    private final Foc file;


    public GpxFileWrapper(Foc f, GpxList l) {
        list = l;
        file = f;

        if (list.getPointList().size()>0)
            setVisibleTrackPoint((GpxPointNode)list.getPointList().getLast());

        this.setVisibleTrackSegment(list.getDelta());
    }

    @Override
    public boolean isLoaded() {
        return true;
    }


    @Override
    public GpxList getGpxList() {
        return list;
    }
    
    @Override
    public Foc getFile() {
        return file;
    }
}

File: aat-lib/src/main/java/ch/bailu/aat_lib/gpx/GpxFileWrapper.kt
package ch.bailu.aat_lib.gpx

import ch.bailu.foc.Foc

class GpxFileWrapper(private val file: Foc, private val list: GpxList) : GpxInformation() {
    init {
        if (list.pointList.size() > 0) {
            val last = list.pointList.last

            if (last is GpxPointNode) {
                setVisibleTrackPoint(last)
            }
        }
        setVisibleTrackSegment(list.getDelta())
    }

    override fun isLoaded(): Boolean {
        return true
    }

    override fun getGpxList(): GpxList {
        return list
    }

    override fun getFile(): Foc {
        return file
    }
}
